### PR TITLE
chore(rust): bump rustc to v1.70.0 (nightly-2023-06-01)

### PR DESCRIPTION
**Pull request by Boshen**
https://blog.rust-lang.org/2023/06/01/Rust-1.70.0.html

### CODE REVIEW

File: crates/rspack_core/src/dependency/code_generatable.rs [hunk 1]
```diff
@@ -257,70 +257,70 @@ pub fn javascript_path_to(
...
-  };
-  ($ast_path:expr, $name:ident($arg:ident: &mut $ty:ident) $b:block) => {
-      $crate::create_css_visitor!(__ $crate::css_path_to(&$ast_path, |n| {
-          matches!(n, swc_core::css::visit::AstParentKind::$ty(_))
-      }), $name($arg: &mut $ty) $b)
-  };
-  (__ $ast_path:expr, $name:ident($arg:ident: &mut $ty:ident) $b:block) => {{
-      struct Visitor<T: Fn(&mut swc_core::css::ast::$ty) + Send + Sync> {
-          $name: T,
+  (visit_mut_stylesheet($arg:ident: &mut Stylesheet) $b:block) => {{
```
Comment (Boshen) [inline crates/rspack_core/src/dependency/code_generatable.rs:260]: The macro broke, fixed by reordering this to the top.

**Review by hyf0 — APPROVED** [review]

File: .cargo/config.toml [hunk 2]
```diff
@@ -26,6 +26,8 @@ rustflags = [
   "-Wclippy::rc_buffer",
   "-Wclippy::rc_mutex",
   "-Wclippy::same_name_method",
+
+  "-Aclippy::default_constructed_unit_structs",
```
Comment (Boshen) [inline .cargo/config.toml:30]:
```
warning: use of `default` to create a unit struct
   --> crates/rspack_binding_options/src/options/raw_output.rs:399:78
    |
399 |           "system" => plugins.push(rspack_plugin_library::SystemLibraryPlugin::default().boxed()),
    |                                                                              ^^^^^^^^^^^ help: remove this call to `default`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#default_constructed_unit_structs

warning: `rspack_binding_options` (lib) generated 7 warnings (run `cargo clippy --fix --lib -p rspack_binding_options` to apply 7 suggestions)
    Checking rspack_testing v0.1.0 (/Users/bytedance/github/rspack/crates/rspack_testing)
warning: use of `default` to create a unit struct
   --> crates/rspack_testing/src/test_config.rs:538:66
    |
538 |           plugins.push(rspack_plugin_library::SystemLibraryPlugin::default().boxed());
    |                                                                  ^^^^^^^^^^^ help: remove this call to `default`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#default_constructed_unit_structs
    = note: `-W clippy::default-constructed-unit-structs` implied by `-W clippy::all`
```

This is fine for us.

File: crates/rspack_core/src/dependency/code_generatable.rs (hunk 1 above)
Comment (IWANABETHATGUY) [reply crates/rspack_core/src/dependency/code_generatable.rs:260]: cc @h-a-n-a PTAL

**Review by ahabhgk — APPROVED** [review]

File: crates/rspack_core/src/dependency/code_generatable.rs (hunk 1 above)
Comment (ahabhgk) [reply crates/rspack_core/src/dependency/code_generatable.rs:260]: looks good to me

**Pull request merged**